indexd_test_utils2/__init__.py

```diff
@@ -0,0 +1,296 @@
+        dbname="indexd_test",
+        version=postgresql_proc.version,
+        password=postgresql_proc.password
+    ):
+        yield setup_database(
+            user=postgresql_proc.user,
+            password=postgresql_proc.password,
+            database="indexd_test",
+            host=f"{postgresql_proc.host}:{postgresql_proc.port}",
+            no_drop=True,
+            no_user=True
+        )
+
+
+def truncate_tables(driver, base):
+    """Drop all the tables in this application's scope.
+
+    This has the same effect as deleting the sqlite file. Your test will have a
+    fresh database for it's run.
+    """
+
+    # Drop tables in reverse order to avoid cascade drop errors.
+    # metadata is a sqlalchemy property.
+    # sorted_tables is a list of tables sorted by their dependencies.
+    with driver.engine.begin() as txn:
+        for table in reversed(base.metadata.sorted_tables):
+            # do not clear schema versions so each test does not re-trigger migration.
+            if table.name not in ["index_schema_version", "alias_schema_version"]:
+                txn.execute(f"TRUNCATE {table.name} CASCADE;")
+
+
+@pytest.fixture
+def index_driver(pg_url):
+    driver = SQLAlchemyIndexDriver(pg_url, auto_migrate=False)
+    yield driver
+    truncate_tables(driver, index_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def alias_driver(pg_url):
+    driver = SQLAlchemyAliasDriver(pg_url, auto_migrate=False)
+    yield driver
+    truncate_tables(driver, alias_base)
+    driver.dispose()
+
+
+@pytest.fixture(scope="session")
+def auth_driver(pg_url):
+    driver = SQLAlchemyAuthDriver(pg_url)
+    yield driver
+    driver.dispose()
+
+
+@pytest.fixture
+def indexd_admin_user(auth_driver):
+    username = password = "admin"
+    auth_driver.add(username, password)
+    yield username, password
+    auth_driver.delete("admin")
+
+
+@pytest.fixture
+def index_driver_no_migrate(pg_url):
+    """
+    This fixture is designed for testing migration scripts and can be used for
+    any other situation where a migration is not desired on instantiation.
+    """
+    driver = SQLAlchemyIndexDriver(pg_url, auto_migrate=False)
+    yield driver
+    truncate_tables(driver, index_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def alias_driver_no_migrate(pg_url):
+    """
+    This fixture is designed for testing migration scripts and can be used for
+    any other situation where a migration is not desired on instantiation.
+    """
+    driver = SQLAlchemyAliasDriver(pg_url, auto_migrate=False)
+    yield driver
+    truncate_tables(driver, alias_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def create_indexd_tables(index_driver, alias_driver, auth_driver):
+    """Make sure the tables are created but don't operate on them directly.
+    Also set up the password to be accessed by the client tests.
+    Migration not required as tables will be created with most recent models
+    """
+    pass
```

Comment:
Is this intentionally a 'pass' or does creating the fixture do something?

Reply:
This fixture calls other fixtures.
Tests who needs `index_driver, alias_driver, auth_driver` just has to call this.